platform/linux-generic/odp_ipsec.c

```diff
@@ -1963,6 +1963,63 @@ int odp_ipsec_out_inline(const odp_packet_t pkt_in[], int num_in,
 	return in_pkt;
 }
 
+int odp_ipsec_sa_info(odp_ipsec_sa_t sa, odp_ipsec_sa_info_t  *sa_info)
+{
+	ipsec_sa_t *ipsec_sa;
+	odp_ipsec_dir_t dir;
+
+	ipsec_sa = _odp_ipsec_sa_entry_from_hdl(sa);
+
+	ODP_ASSERT(ipsec_sa != NULL);
+	ODP_ASSERT(sa_info != NULL);
+
+	memset(sa_info, 0, sizeof(*sa_info));
+
+	dir = ipsec_sa->param.dir;
+
+	sa_info->param = (odp_ipsec_sa_param_t)ipsec_sa->param;
```

Comment:
The cast is unnecessary. ipsec_sa->param is of odp_ipsec_sa_param_t type.

Reply:
Agreed.